apps/rig_outbound_gateway/lib/rig_outbound_gateway/kafka/group_subscriber.ex

```diff
@@ -114,13 +114,14 @@ defmodule RigOutboundGateway.Kafka.GroupSubscriber do
   If it exceedes maximum timeout returns :error, otherwise stops and returns :ok as
   soon as possible.
   """
+  def wait_for_consumer_ready(topic, partition, timeout \\ 10_000)
   @spec wait_for_consumer_ready(String.t(), integer, integer) :: {:error, String.t()}
   def wait_for_consumer_ready(topic, partition, timeout) when timeout <= 0 do
     {:error, "Consumer ready check for topic: #{topic} and partition: #{partition} timeouted."}
   end
 
   @spec wait_for_consumer_ready(String.t(), integer, integer) :: {:ok, String.t()}
-  def wait_for_consumer_ready(topic, partition, timeout \\ 10_000) do
+  def wait_for_consumer_ready(topic, partition, timeout) do
```

Comment:
just a minor comment about the two specs: having one spec per function clause is optional and makes perfect sense if there something like "if I put a String.t in there, I get a String.t out, but if I put an atom in there, it'll be an atom" (you get the idea). However, in this case I feel it would be clearer (and less to write/read/maintain) to have just one combined spec, as I don't see any gain in expressing the same thing using two lines:
```elixir
@spec wait_for_consumer_ready(String.t(), integer, integer) :: {:ok | :error, String.t()}
```
What do you think?

PS: I think `integer` here should actually be `non_neg_integer()` in both cases